### PR TITLE
Remove incorrect string quoting of '+' for string concatenation

### PR DESCRIPTION
**Pull request by henrikt-ma**

### CODE REVIEW

**Review by HansOlsson — COMMENTED** [review]
I believe that the underlying issue here is that it is a strange description, and to me the quoting is actually kind of correct if we assume that it is the grammar item + (we are discussing the grammar after all) and not the Modelica item +. 

However, what we want to describe is that to concatenate two constant strings you write `"a"+"b"`; but we don't include that in the text and instead have something that isn't Modelica and then describe that it's different in Modelica without fully writing the Modelica variant.

Something like:

> String constants can be concatenated so that \lstinline!"a" + "b"! becomes \lstinline!"ab"!.

Completely skipping the comparison with C; and having inline expressions; or possibly adding comparison with C:

> String constants can be concatenated so that \lstinline!"a" + "b"! becomes \lstinline!"ab"!; there is no support for the C-style of \lstinline!"a" "b"! becoming \lstinline!"ab"!.

**Comment by henrikt-ma** [comment]
> I believe that the underlying issue here is that it is a strange description, and to me the quoting is actually kind of correct if we assume that it is the grammar item + (we are discussing the grammar after all) and not the Modelica item +.

Sure, switching to `language=grammar` would make it technically correct with the double quotes, but hardly more readable.  Seen that way, the first two strings should probably have `language=C`…

> However, what we want to describe is that to concatenate two constant strings you write `"a"+"b"`; but we don't include that in the text and instead have something that isn't Modelica and then describe that it's different in Modelica without fully writing the Modelica variant.

Right.

> Something like:
> 
> > String constants can be concatenated so that \lstinline!"a" + "b"! becomes \lstinline!"ab"!.

Without the relation to the C language, I don't see how this statement fits in this chapter.  It is the comparison to C, where this concatenation happens in some strange low level translation phase, that this is worth mentioning at all.

> Completely skipping the comparison with C; and having inline expressions; or possibly adding comparison with C:
> 
> > String constants can be concatenated so that \lstinline!"a" + "b"! becomes \lstinline!"ab"!; there is no support for the C-style of \lstinline!"a" "b"! becoming \lstinline!"ab"!.

How about this, emphasizing that what we provide is just a normal binary expression?
> Concatenation of string literals requires a binary expression.  For example, \lstinline!"a" + "b"! becomes \lstinline!"ab"!. There is no support for the C/C++ style of concatenating adjacent string literal tokens (for example, \lstinline[language=C]!"a" "b"! becoming \lstinline[language=C]!"ab"!).

**Comment by HansOlsson** [comment]
> How about this, emphasizing that what we provide is just a normal binary expression?
> 
> > Concatenation of string literals requires a binary expression.  For example, \lstinline!"a" + "b"! becomes \lstinline!"ab"!. There is no support for the C/C++ style of concatenating adjacent string literal tokens (for example, \lstinline[language=C]!"a" "b"! becoming \lstinline[language=C]!"ab"!).

Yes, makes sense.

File: chapters/syntax.tex [hunk 1]
```diff
@@ -48,8 +48,7 @@ \section{Lexical conventions}\label{lexical-conventions}
   Q-IDENT where they are treated as part of the STRING or Q-IDENT
   lexical unit.
 \item
-  String constant concatenation \lstinline!"a" "b"! becoming \lstinline!"ab"! (as in C) is
-  replaced by the \lstinline!"+"! operator in Modelica.
+  String constant concatenation \lstinline!"a" "b"! becoming \lstinline!"ab"! (as in C) is replaced by the \lstinline!+! operator in Modelica.
```
Comment (henrikt-ma) [inline chapters/syntax.tex:51]: As suggested in conversation with @HansOlsson:
```suggestion
  Concatenation of string literals requires a binary expression. For example, \lstinline!"a" + "b"! becomes \lstinline!"ab"!.
  There is no support for the C/C++ style of concatenating adjacent string literal tokens (for example, \lstinline[language=C]!"a" "b"! becoming \lstinline[language=C]!"ab"!).
```

**Comment by henrikt-ma** [comment]
> Yes, makes sense.

Done, except saying _evaluates to_ about the binary expression.

**Review by HansOlsson — APPROVED** [review]
Looks good.

**Pull request merged**